packages/grid/index.ts

```diff
@@ -37,32 +27,32 @@ export const grid = starryComponent<HTMLElement>(function (
  traits: StarryUITrait[]
 ) {
  const { height, width } =
-  traits.find((x) => x.type === 'cellRender') ?? defaultCellDimensionsTrait
+  traits.find((x) => x.type === 'cellDimensions') ?? defaultCellDimensionsTrait
 
  const { cellRender } =
   traits.find((x) => x.type === 'cellRender') ?? defaultCellRenderTrait
 
  return function (config?: StarryUITraitConfig) {
-  let elem = document.createElement(config?.tagName ?? 'div')
+  const elem = document.createElement(config?.tagName ?? 'div')
   applyTraits(elem, traits, Object.assign({}, defaultConfigConfig, config))
 
-  elem = createGrid({
+  const grid = createGrid({
    cellHeight: height,
    cellWidth: width,
    renderCell: cellRender,
   }).container
 
-  return elem
+  return elem.appendChild(grid)
  }
 })
 
-interface Params {
+interface CreateGridParams {
  cellWidth: number
  cellHeight: number
  renderCell(x: number, y: number): HTMLElement
 }
 
-function createGrid({ cellHeight, cellWidth, renderCell }: Params) {
+function createGrid({ cellHeight, cellWidth, renderCell }: CreateGridParams) {
  const container = document.createElement('div')
  container.style.position = 'relative'
```

Comment:
🟨 This is where we need to use the theme facets to add styles to the element. We need the grid to be able to fill up all the available space, with `width: 100%, height: 100%`. We want to use them facets to achieve this.

Here's how we can use the Starry UI trait system to attach these styles:

```
const gridContainerConfig: StarryUITraitConfig = { themeFacet: 'grid' }
applyTraits(container, traits, gridContainerConfig)
```

Note that `traits` comes from the root level of the component, this is so the inside parts of the component use the same theme trait as the root level, the styles from the same theme will be applied to both elements, but using the correct theme facet for each HTML element.

Next, we need to add a ResizeObserver to update the grid when the size changes. See https://web.dev/articles/resize-observer for instructions

Reply:
Can you clarify how to attach the style to the starry UI trait?

Reply:
🟨 @Vikulakav yes, look at the `frame` package as an example. Then look at theme-midnight to see the CSS. We're using a css-in-js solution here, which allows nesting and complex states like hover and active, you can read the file and understand how the CSS theming works. You can switch the theme when running `npm run docs:watch` in the top right corner menu in the running application.

See https://github.com/starryui/starryui/blob/main/packages/theme-midnight/index.ts#L180 :

```TypeScript
  frame: {
   border: '1px solid var(--theme4)',
   borderRadius: 'var(--dimension2)',
   boxSizing: 'border-box',
   height: '100%',
   overflowX: 'hidden',
   overflowY: 'auto',
   position: 'relative',
   width: '100%',
  },
```

So you can add a `grid` facet right after `frame`. Don't forget to also update the other theme, `theme-sandstone`